Update Type field of transaction filters - Closes #2560

colado opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

### What issue have I solved?
<!--- Complementary description if needed -->
#2560

### How have I implemented/fixed it?
- Created `DropdownFilter` inside `FilterDropdownButton`.
- Modified `Select` component to accept placeholders.
- Applied `DropdownFilter` to Type field.

### How has this been tested?
Made sure `Type` filter dropdown aligned with [design](https://projects.invisionapp.com/d/main?origin=v7#/console/18539694/389102939/preview).
It was decided to keep transaction naming consistent across the app so the type names are the ones used on the [constants file](https://github.com/LiskHQ/lisk-hub/blob/development/src/constants/transactionTypes.js) and not on the design preview.


### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)

slaweet commented on src/components/shared/filterDropdownButton/index.js:
```diff
@@ -10,13 +10,15 @@ import IntegerFilter from './integerFilter';
   text: TextFilter,
   integer: IntegerFilter,
   address: AddressFilter,
+  dropdown: DropdownFilter,
```
💅 Given `DropdownFilter` uses `Select` component and not `Dropdown` component, I suggest renaming for consistency.
```suggestion
  select: SelectFilter,
```

colado replied:
✅

slaweet commented on src/components/toolbox/select/index.js:
```diff
@@ -49,7 +51,8 @@ class Select extends React.Component {
             readOnly
-            value={options[selected].label}
+            placeholder={placeholder}
+            value={typeof selected === 'number' ? options[selected].label : ''}
```
💅  IMO it would be more readable and also safer (it also ensures `selected < options.length`) to say:
```suggestion
            value={options[selected] ? options[selected].label : ''}
```

colado replied:
✅

slaweet commented on src/components/toolbox/select/index.js:
```diff
@@ -62,7 +65,7 @@ class Select extends React.Component {
           {options.map((option, index) => (
             <span
-              className={`${styles.option} option`}
+              className={`${smallSizeOptions ? styles.optionSmall : styles.optionLarge} option`}
```
```suggestion
              className={['option', styles.option, styles[size]].join(' ')}
```
💅 `smallSizeOptions` prop seems duplicate to `size` prop. IMO options should be sized based on `size`, similar to how it's handled in 
https://github.com/LiskHQ/lisk-hub/blob/8b610538bdf9bb4e6345cefff24c997eb91c400b/src/components/toolbox/buttons/css/base.css#L14-L36
and 
https://github.com/LiskHQ/lisk-hub/blob/8b610538bdf9bb4e6345cefff24c997eb91c400b/src/components/toolbox/buttons/button.js#L32-L37

colado replied:
I have added a class for `.xs` and applied the size to it. Let me know if you have any other suggestion.

slaweet commented on src/components/shared/filterDropdownButton/dropdownFilter.js:
```diff
@@ -0,0 +1,43 @@
+  label, t, placeholder, filters, name, updateCustomFilters,
+}) => {
+  const transactionTypes = Object.keys(transactionNames(t))
+    .filter((tx, i) => i <= 4)
```
```suggestion
    .slice(0, 5)
```
whould be simpler 💅

colado replied:
✅

slaweet commented on src/components/shared/filterDropdownButton/dropdownFilter.js:
```diff
@@ -0,0 +1,43 @@
+      [name]: {
+        value,
+        error: '',
+        loading: false,
```
Are these two needed?

colado replied:
No, removed!

slaweet commented on src/components/shared/transactionsTable/index.js:
```diff
@@ -149,7 +150,7 @@ class TransactionsTable extends React.Component {
                         tooltipClassName={styles.tooltip}
                         content={<LiskAmount val={transaction.fee} token={tokenMap.LSK.key} />}
```
```suggestion
                         content={<LiskAmount val={transaction.fee} token={tokenMap.LSK.key} />}
                         size="s"
```
I was adding size (to align tooltip padding) in https://github.com/LiskHQ/lisk-hub/commit/d0077e97bc165dda1feb37e2de3e12dd2fce9f4b#diff-2e0c48c8b82d2186028ea2f9a9286d29R79 but looks like it got lost in some merge conflict
https://github.com/LiskHQ/lisk-hub/blob/4948dc3601820ad487007bbdc9ea51366a442ec6/src/components/screens/monitor/blocks/blocks.js#L96

🎨 now it looks a bit weird
![Screenshot 2019-10-18 at 12 31 14](https://user-images.githubusercontent.com/1254342/67087844-09089a00-f1a4-11e9-8366-34b91b7bc146.png)

colado replied:
After discussing it with Yashar, I have increased the size of the fee tooltip and make it show on the left, that way the end of the table doesn't overlap with the tooltip and it also aligns with the status behaviour one. I have added this as a different size to the `Tooltip` component. Let me know if this looks ok to you.

colado commented:
Regarding the 'Clear all filters' bug I'm going to wait  to see which solution design proposes to removing the `Type` filter to the initial value, since it may affect this, and implement the fix then.

colado commented:
@slaweet the new design for deselecting the type is ready here: https://projects.invisionapp.com/share/ZRU30GY5NTX/#/screens/389102939

I have implemented those changes and fixed the bug that didn't clear the type selection after clearing the filters.

slaweet commented on src/components/toolbox/select/index.js:
```diff
@@ -11,13 +11,23 @@ class Select extends React.Component {
+      this.setState({ selected: this.props.selected });
+    }
+  }
+
```
💅 I think `this.props.selected` can be used directly instead of mirorring it to `this.state.selected`

colado replied:
I removed the internal `selected` state and refactored the component so that it can work with the new styles in the rest of the instances.

Efefefef commented:
It is not clear if transaction type names should be all words uppercase like in design or not.
